nmostesting/mocks/Registry.py

```diff
@@ -271,3 +417,198 @@ def heartbeat(version, node_id):
         return jsonify({"health": int(time.time())})
     else:
         abort(404)
+
+
+@REGISTRY_API.route('/x-nmos/query', methods=["GET"], strict_slashes=False)
+def query_root():
+    registry = REGISTRIES[flask.current_app.config["REGISTRY_INSTANCE"]]
+    if not registry.enabled:
+        abort(503)
+    authorized = registry.check_authorized(request.headers, request.path)
+    if authorized is not True:
+        abort(authorized)
+
+    base_data = ['v1.0/', 'v1.1/', 'v1.2/', 'v1.3/']
+
+    return Response(json.dumps(base_data), mimetype='application/json')
+
+
+@REGISTRY_API.route('/x-nmos/query/<version>', methods=["GET"], strict_slashes=False)
+def query(version):
+    registry = REGISTRIES[flask.current_app.config["REGISTRY_INSTANCE"]]
+    if not registry.enabled:
+        abort(503)
+    authorized = registry.check_authorized(request.headers, request.path)
+    if authorized is not True:
+        abort(authorized)
+
+    base_data = ['devices/', 'flows/', 'nodes/', 'receivers/', 'senders/', 'sources/', 'subscriptions/']
+
+    return Response(json.dumps(base_data), mimetype='application/json')
+
+
+def compare_resources(resource1, resource2):
+    try:
+        return NMOSUtils.compare_resource_version(resource1['version'], resource2['version'])
+    except Exception as e:
+        print(e)
+        return 0
+
+
+@REGISTRY_API.route('/x-nmos/query/<version>/<resource>', methods=["GET"], strict_slashes=False)
+def query_resource(version, resource):
+    registry = REGISTRIES[flask.current_app.config["REGISTRY_INSTANCE"]]
+    if not registry.enabled:
+        abort(503)
+    authorized = registry.check_authorized(request.headers, request.path)
+    if authorized is not True:
+        abort(authorized)
+
+    MIN_SINCE = "0:0"
+    MAX_UNTIL = NMOSUtils.get_TAI_time()
+
+    base_data = []
+
+    if request.args.get('paging.since') or request.args.get('paging.until'):
+        registry.pagination_used = True
+
+    since = request.args.get('paging.since') if request.args.get('paging.since') else MIN_SINCE
+    until = request.args.get('paging.until') if request.args.get('paging.until') else MAX_UNTIL
+    new_until = until
+
+    try:
+        valid_resource_types = ['device', 'flow', 'node', 'receiver', 'sender', 'source', 'subscription']
```

Comment:
Does this actually work for `/subscriptions`?

Reply:
Refactored so subscriptions are now resources, and this endpoint now works